src/commands/recover.py

```diff
@@ -0,0 +1,245 @@
+import asyncio
+from dataclasses import dataclass, field
+from pathlib import Path
+
+import click
+from eth_typing import HexAddress
+from eth_utils import add_0x_prefix
+from tqdm import tqdm
+from web3.types import HexStr
+
+from src.commands.validators_exit import EXITING_STATUSES
+from src.common.clients import consensus_client, execution_client
+from src.common.contracts import vault_contract
+from src.common.credentials import CredentialManager
+from src.common.password import generate_password, get_or_create_password_file
+from src.common.utils import log_verbose
+from src.common.validators import validate_eth_address, validate_mnemonic
+from src.common.vault_config import VaultConfig
+from src.config.settings import AVAILABLE_NETWORKS, GOERLI, settings
+
+
+@dataclass
+class RegisteredValidator:
+    index: int
+    public_key: HexStr
+    status: str
+    keystore_found: bool = field(default=False)
```

Comment:
Looks like index and keystore_found is not used. Do we actually need a dataclass?